internal/agent/device/integrity/exporter.go

```diff
@@ -0,0 +1,31 @@
+package integrity
+
+import (
+	"context"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/agent/device/status"
+	"github.com/flightctl/flightctl/internal/tpm"
+)
+
+var _ status.Exporter = (*Exporter)(nil)
+
+type Exporter struct {
+	tpmClient *tpm.Client
+}
+
+func NewExporter(tpmClient *tpm.Client) *Exporter {
+	return &Exporter{
+		tpmClient: tpmClient,
+	}
+}
+
+func (e *Exporter) Status(ctx context.Context, deviceStatus *v1alpha1.DeviceStatus, opts ...status.CollectorOpt) error {
```

Comment:
note: integrity status is owned by the service not the agent for security purposes the agent can not declare itself as verified the service should ensure.